src/SprykerEco/Yves/Heidelpay/Plugin/Router/HeidelpayRouteProviderPlugin.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Yves\Heidelpay\Plugin\Router;
+
+use Spryker\Yves\Router\Plugin\RouteProvider\AbstractRouteProviderPlugin;
+use Spryker\Yves\Router\Route\RouteCollection;
+
+class HeidelpayRouteProviderPlugin extends AbstractRouteProviderPlugin
+{
+    protected const HEIDELPAY_BUNDLE_NAME = 'Heidelpay';
+    protected const HEIDELPAY_PAYMENT = 'heidelpay-payment';
+    protected const HEIDELPAY_PAYMENT_FAILED = 'heidelpay-payment-failed';
+    protected const HEIDELPAY_EASY_CREDIT_PAYMENT = 'heidelpay-easy-credit-payment';
+    protected const HEIDELPAY_EASY_CREDIT_INITIALIZE_PAYMENT = 'heidelpay-easy-credit-initialize-payment';
+    protected const HEIDELPAY_IDEAL_AUTHORIZE = 'heidelpay-ideal-authorize';
+    protected const HEIDELPAY_CREDIT_CARD_REGISTER = 'heidelpay-cc-register';
+    protected const HEIDELPAY_CREDIT_CARD_REGISTER_SUCCESS = 'heidelpay-cc-register-success';
+    protected const HEIDELPAY_DIRECT_DEBIT_REGISTER = 'heidelpay-dd-register';
+    protected const HEIDELPAY_DIRECT_DEBIT_REGISTER_SUCCESS = 'heidelpay-dd-register-success';
+    protected const HEIDELPAY_NOTIFICATION = 'heidelpay-notification';
+
+    /**
+     * @param \Spryker\Yves\Router\Route\RouteCollection $routeCollection
+     *
+     * @return \Spryker\Yves\Router\Route\RouteCollection
+     */
+    public function addRoutes(RouteCollection $routeCollection): RouteCollection
+    {
+        $routeCollection = $this->addHeidelpayPaymentRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayPaymentFailedRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayEasyCreditPaymentRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayEasyCreditInitializePaymentRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayIdealAuthorizeRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayCreditCardRegisterRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayCreditCardRegisterSuccessRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayDirectDebitRegisterRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayDirectDebitRegisterSuccessRoute($routeCollection);
+        $routeCollection = $this->addHeidelpayNotificationRoute($routeCollection);
+
+        return $routeCollection;
+    }
+
+    /**
+     * @param \Spryker\Yves\Router\Route\RouteCollection $routeCollection
```

Comment:
For each `add...Route` method please add `@uses` annotation that links used controller

Example:

```
    /**
     * @uses \SprykerShop\Yves\AgentPage\Controller\AgentController::indexAction()
     *
     * @param \Spryker\Yves\Router\Route\RouteCollection $routeCollection
     *
     * @return \Spryker\Yves\Router\Route\RouteCollection
     */
    protected function addOverviewRoute(RouteCollection $routeCollection): RouteCollection
```

Reply:
Fixed only partially, please check comment one more time.

Reply:
also added action methods